api/providers/tasks.py

```diff
@@ -99,6 +107,213 @@ def prepare_for_registration_bulk_creation(payload_hash, initiator_id, provider_
+        except AbstractNode.DoesNotExist:
+            logger.error('Nod does not exist: [node_id={}]'.format(node_id))
+            error_list.append(row_external_id)
+            return False
+        except AbstractNode.MultipleObjectsReturned:
+            logger.error('Multiple nodes returned: [node_id={}]'.format(node_id))
+            error_list.append(row_external_id)
+            return False
+
+    # Subject
+    subject_names = metadata.get('subjects', [])
+    all_subjects = []
+    for subject_name in subject_names:
+        subject_list = provider.all_subjects.filter(text=subject_name)
+        if not subject_list:
+            logger.error('Subject not found: [subject_name={}]'.format(subject_name))
+            error_list.append(row_external_id)
+            return False
+        if len(subject_list) > 1:
+            logger.error('Duplicate subjects found: [subject_name={}]'.format(subject_name))
+            error_list.append(row_external_id)
+            return False
+        subject = subject_list.first()
+        all_subjects.append(subject)
+        parent_subject = subject.parent
+        while parent_subject:
+            all_subjects.append(parent_subject)
+            parent_subject = parent_subject.parent
+
+    # Node License
+    license_name = metadata.get('license').get('name')
+    try:
+        node_license = NodeLicense.objects.get(name=license_name)
+    except NodeLicense.DoesNotExist:
+        logger.error('License not found: [license_name={}]'.format(license_name))
+        error_list.append(row_external_id)
+        return False
+
+    # Editable fields
+    data = {
+        'title': metadata.get('title'),
+        'category': metadata.get('category').lower(),
+        'description': metadata.get('description'),
+        # TODO: handle licenses that doesn't need the "year" and "copyrightHolders" properties
+        'node_license': {
+            'id': node_license.license_id,
+            'year': metadata.get('license').get('required_fields').get('year'),
+            'copyright_holders': metadata.get('license').get('required_fields').get('copyright_holders'),
+        },
+    }
+
+    # Creating the draft registration
+    draft = DraftRegistration.create_from_node(
+        initiator,
+        schema,
+        node=node,
+        data=data,
+        provider=provider
+    )
+    row.draft_registration = draft
+    row.save()
+    # TODO: remove initiator from bibliographic/citation authors
+
+    # Set subjects
+    # TODO: switch use `.set_subject()` from class `TaxonomizableMixin`
+    for subject in all_subjects:
+        draft.subjects.add(subject)
+        draft.save()
+
+    # Set affiliated institutions
+    # TODO: switch to use `.add_affiliated_institution()` from `AffiliatedInstitutionMixin`
+    institution_names = metadata.get('affiliated institutions')
+    for name in institution_names:
+        try:
+            institution = Institution.objects.get(name=name, is_delete=False)
```

Comment:
Maybe we should only allow institutions with which the initiator are affiliated to be added as institutions for the draft. So if you use `add_affiliated_institution()` it should take care of that and raise exception if the initiator is not affiliated with any of the institutions.

Reply:
Yes, this is also in my to-do list.

Reply:
Done. As discussed, we will need to update this affiliation to allow institutions from all contributors in the future.